amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/job/DispositionLifecycleJobExecuter.java

```diff
@@ -214,7 +215,14 @@ public void executeImpl()
 
                 // execute search
                 ResultSet results = searchService.query(params);
-                List<NodeRef> resultNodes = results.getNodeRefs();
+                if(results != null)
+                {
+                    // filtering out the hold/freezed cases from the result set
+                    resultNodes =
+                            results.getNodeRefs().stream().filter(node -> nodeService.getPrimaryParent(node) == null ?
```

Comment:
How is it possible for `nodeService.getPrimaryParent(node) == null`?

Reply:
I'm not aware of the scenario but when we are removing the null check then the Test cases are getting failed with HeapMemory Issue.